MastodonSDK/Sources/MastodonCore/FetchedResultsController/StatusDataController.swift

```diff
@@ -35,39 +39,100 @@ public final class StatusDataController {
+        var newRecords = Array(records)
+        guard let index = newRecords.firstIndex(where: { $0.id == status.id }) else {
+            logger.warning("\(Self.entryNotFoundMessage)")
+            return
+        }
+        newRecords[index] = status.inheritSensitivityToggled(from: newRecords[index])
+        records = newRecords
+    }
+    
+    @MainActor
+    private func updateBookmarked(_ status: MastodonStatus, _ isBookmarked: Bool) {
+        var newRecords = Array(records)
+        guard let index = newRecords.firstIndex(where: { $0.id == status.id }) else {
+            logger.warning("\(Self.entryNotFoundMessage)")
+            return
+        }
+        newRecords[index] = status.inheritSensitivityToggled(from: newRecords[index])
+        records = newRecords
+    }
+    
+    @MainActor
+    private func updateFavorited(_ status: MastodonStatus, _ isFavorited: Bool) {
+        var newRecords = Array(records)
+        if let index = newRecords.firstIndex(where: { $0.id == status.id }) {
+            // Replace old status entity
+            let existingRecord = newRecords[index]
+            let newStatus = status.inheritSensitivityToggled(from: existingRecord)
+            newRecords[index] = newStatus
+        } else if let index = newRecords.firstIndex(where: { $0.reblog?.id == status.id }) {
+            // Replace reblogged entity of old "parent" status
+            let existingRecord = newRecords[index]
+            let newStatus = status.inheritSensitivityToggled(from: existingRecord)
+            newStatus.reblog = status
+            newRecords[index] = newStatus
+        } else {
+            logger.warning("\(Self.entryNotFoundMessage)")
+        }
+        records = newRecords
+    }
+    
+    @MainActor
+    private func updateReblogged(_ status: MastodonStatus, _ isReblogged: Bool) {
         var newRecords = Array(records)
-        for (i, record) in newRecords.enumerated() {
-            if record.id == status.id {
-                newRecords[i] = status
-            } else if let reblog = status.reblog, reblog.id == record.id {
-                newRecords[i] = status
-            } else if let reblog = record.reblog, reblog.id == status.id {
-                // Handle reblogged state
-                let isRebloggedByAnyOne: Bool = records[i].reblog != nil
 
-                let newStatus: MastodonStatus
-                if isRebloggedByAnyOne {
-                    // if status was previously reblogged by me: remove reblogged status
-                    if records[i].entity.reblogged == true && status.entity.reblogged == false {
-                        newStatus = .fromEntity(status.entity)
-                    } else {
-                        newStatus = .fromEntity(records[i].entity)
-                    }
-                    
-                } else {
-                    newStatus = .fromEntity(status.entity)
-                }
+        switch isReblogged {
```

Comment:
Using a `switch` with a `Bool` seems creative to me, it made me laugh out loud :D But I guess it makes sense with those `guard`s